libminifi/src/c2/C2Agent.cpp

```diff
@@ -747,6 +752,20 @@ void C2Agent::handle_update(const C2ContentResponse &resp) {
   }
 }
 
+/**
+ * Updates a property
+ */
+bool C2Agent::update_property(const std::string &property_name, const std::string &property_value, bool persist) {
+  if (update_service_->canUpdate(property_name)) {
+    configuration_->set(property_name, property_value);
+    if (persist) {
+      configuration_->persistProperties();
+      return true;
+    }
+  }
+  return false;
```

Comment:
This looks a bit strange.
Why does this function always return false in case persist=false?

Reply:
because it wasn't persisted and that's the semantics